pkg/targetgroupbinding/targets_manager.go

```diff
@@ -145,6 +154,60 @@ func (m *cachedTargetsManager) ListTargets(ctx context.Context, tgARN string) ([
 	return cloneTargetInfoSlice(refreshedTargets), nil
 }
 
+// A copy of ListTargets, except ListOwnedTargets requires an EKSInfoResolver so that it can only return targets that
+// are owned by the cluster. It used CIDRs to determine if an IP target is owned by the cluster, as assumes instances are
+func (m *cachedTargetsManager) ListOwnedTargets(ctx context.Context, tgARN string, eksInfoResolver networking.EKSInfoResolver) ([]TargetInfo, error) {
+	m.targetsCacheMutex.Lock()
+	defer m.targetsCacheMutex.Unlock()
+
+	if rawTargetsCacheItem, exists := m.targetsCache.Get(tgARN); exists {
+		targetsCacheItem := rawTargetsCacheItem.(*targetsCacheItem)
+		targetsCacheItem.mutex.Lock()
+		defer targetsCacheItem.mutex.Unlock()
+		refreshedTargets, err := m.refreshUnhealthyTargets(ctx, tgARN, targetsCacheItem.targets)
+		if err != nil {
+			return nil, err
+		}
+		targetsCacheItem.targets = refreshedTargets
+		return cloneTargetInfoSlice(refreshedTargets), nil
```

Comment:
There is a slight bug here, when we use take targets from the cache we do not perform filtering and end up regressing on the multi-cluster behavior.